REST/c4Listener.cc

```diff
@@ -38,6 +38,40 @@ string C4Listener::URLNameFromPath(slice pathSlice) {
     return Listener::databaseNameFromPath(FilePath(pathSlice, ""));
 }
 
+namespace {
+    std::stringstream& operator<<(std::stringstream& ss, const C4ListenerConfig& config) {
+        ss << "{" << "apis: " << (config.apis == kC4RESTAPI ? "REST" : "Sync") << ", "
+        << "networkInterface: " << std::string(config.networkInterface) << ", "
+        << "port: " << config.port << ", "
+        << "tlsConfig: " << "{";
+        if (config.tlsConfig != nullptr) {
+            ss << "privateKeyRepresentation: " <<
+                (config.tlsConfig->privateKeyRepresentation == kC4PrivateKeyFromCert
+                 ? "PrivateKeyFromCert" : "PrivateKeyFromKey") << ", "
+            << "key: " << (config.tlsConfig->key == nullptr ? NULL : "***") << ", "
+            << "certificate: " << (config.tlsConfig->certificate == nullptr ? "NULL" : "***") << ", "
+            << "requireClientCerts: " << config.tlsConfig->requireClientCerts << ", "
+            << "rootClientCerts: " << (config.tlsConfig->rootClientCerts == nullptr ? "NULL" : "***") << ", "
+            << "certAuthCallback: " << (config.tlsConfig->certAuthCallback == nullptr ? "NULL" : "***") << ", "
+            << "tlsCallbackContext: " << (config.tlsConfig->tlsCallbackContext == nullptr ? "NULL" : "***");
+        }
+        ss << "}, "
+        << "httpAuthCallback: " << (config.httpAuthCallback == nullptr ? "NULL" : "***") << ", "
+        << "callbackContext: " << (config.callbackContext == nullptr ? "NULL" : "***") << ", "
+        << "directory: " << std::string(config.directory) << ", ";
```

Comment:
If config.directory is nullslice, I think we should put NULL here.

Reply:
same above